.github/workflows/testsuite.yml

```diff
@@ -0,0 +1,123 @@
+name: testsuite
+
+on:
+  push:
+    branches:
+      - "*"
+    tags-ignore:
+      - "*"
+  pull_request:
+
+jobs:
+  ubuntu:
+    env:
+      PERL_USE_UNSAFE_INC: 0
+      AUTHOR_TESTING: 1
+      AUTOMATED_TESTING: 1
+      RELEASE_TESTING: 1
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: perl -V
+        run: perl -V
+      - name: Makefile.PL
+        run: perl -I$(pwd) Makefile.PL
+      - name: make test
+        run: make test
+
+  linux:
+    name: "linux ${{ matrix.perl-version }}"
+    needs: [ubuntu]
+    env:
+      PERL_USE_UNSAFE_INC: 0
+      AUTHOR_TESTING: 1
+      AUTOMATED_TESTING: 1
+      RELEASE_TESTING: 1
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        perl-version:
+          [
+            "perl:5.32",
+            "perl:5.30",
+            "perl:5.28",
+            "perl:5.26",
+            "perl:5.24",
+            "perl:5.22",
+            "perl:5.20",
+            "perl:5.18",
+            "perl:5.16",
+            "perl:5.14",
+            "perl:5.12",
+            "perl:5.10",
+            # currently broken on 5.8 and 5.6
```

Comment:
5.8 and 5.6 should work now, so these can be uncommented :-)

Reply:
no this does not work for now, let's fix them later?

Reply:
would you prefer merging the workflow enabling 5.8 and 5.6?
so we this would expose the issues?
I was considering having a clean status then work on 5.8 and 5.6 to pass
they have different issues to fix... 

for example one of the issue with 5.6 is the lack of parent.pm which is a dependency